### PR TITLE
Make sure NearInteraction triggers CanInteract check

### LINKED ISSUE #8479: Interactable's CanDeselect does not have any effect with NearInteraction
## Describe the bug

Interactable toggle buttons that have CanDeselect property set to false can still be deselected using Near Interaction.

## To reproduce

1. Open InteractableExamples scene
2. Change CanDeselect property inside ToggleButton object's Interactable
3. Hit play
4. Select toggle button either by far or near interaction
5. Use near interaction to deselect the button
6. Repeat the procedure and try to deselect with far interaction - that won't work

## Expected behavior

When CanDeselect is set to false it should not allow deselection, no matter what input is used. 
Near interaction and far interaction should always result in the same effect, right now their functionality is different,
as user can freely deselect radio and toggle buttons.

## Your setup (please complete the following information)

- Unity Version 2019.4.0f1
- MRTK Version 2.4.0

## Target platform (please complete the following information)

- HoloLens 2
- UWP
- IL2CPP

## Additional context

After putting those lines:
`
            if (!CanInteract())
            {
                return;
            }
`
inside TriggerOnClick function in Interactable.cs, line 1238, issue disappears. It seems that PhysicalPressEventRouter omits this check.

### PR DESCRIPTION
**Pull request by MaxWang-MS**
## Overview
Interactable should always check for `CanInteract` no matter it is responding to far interaction or near interaction.

Thanks @rfurmaniak for proposing the change!

## Changes
- Fixes #8479

### CODE REVIEW

**Comment by davidkline-ms** [comment]
We are at the point in the (2.5) release where we are only fixing regressions. We will merge this into the development branch (for 2.6)

**Review by davidkline-ms — COMMENTED** [review]
overall looks good. one small question

File: Assets/MRTK/SDK/Features/UX/Interactable/Scripts/InteractableToggleCollection.cs [hunk 1]
```diff
@@ -106,7 +106,7 @@ public void SetSelection(int index, bool force = false, bool fireOnClick = false
 
                 if (fireOnClick)
                 {
-                    ToggleList[index].TriggerOnClick();
+                    ToggleList[index].TriggerOnClick(true);
```
[the remark this answers is not in the document]
Comment (MaxWang-MS) [reply Assets/MRTK/SDK/Features/UX/Interactable/Scripts/InteractableToggleCollection.cs:110]: Yes, the intent here is to ignore the CanInteract check. Please take a look at the new comment above that line and let me know if things are clearer now.

**Review by davidkline-ms — APPROVED** [review]

**Pull request merged**